### ISSUE TITLE
Issue about using the nvhpc compiler on Casper with NUOPC driver

### ISSUE THREAD

**Issue by sjsprecious** [ESMCI/ccs_config_cesm#11]
NUOPC driver requires the ESMF library to be prebuilt. On Casper, the ESMF library is available for `nvhpc/21.11` and `nvhpc/22.1` with `openmpi/4.1.1`. However, I encountered the following runtime error with both nvhpc compiler versions when using the NUOPC driver:

`/glade/scratch/sunjian/cam6_run/F2000climo.f19_f19_mg17.casper.nvhpc-gpu.gpu02_pcols00384_mpi036_nuopc/bld/cesm.exe: symbol lookup error: /glade/u/apps/dav/opt/openmpi/4.1.1/nvhpc/22.1/lib/openmpi/mca_op_avx.so: undefined symbol: llvm.x86.avx512.pmins.d`

There seems to be a bug in the nvhpc compiler regarding the AVX instruction (https://github.com/open-mpi/ompi/issues/9444). Is it used somewhere in the ESMF lib or NUOPC driver?

**Comment by johnmauff** [comment]
@jedwards4b @mvertens hopefully we can get an update on this issue at our Feb 14th meeting.

**Comment by cponder** [comment]
You could get this kind of error if the library was built with the wrong compiler.
What are the paths to the nvhpc & esmf libraries that you're using?

**Comment by sjsprecious** [comment]
Thanks @cponder . The paths to nvhpc & esmf modules on Casper are:
NVHPC: `/glade/u/apps/opt/nvhpc/21.11`
ESMF: `/glade/p/cesmdata/cseg/PROGS/esmf/8.3.0b06_casper/openmpi/4.1.1/nvhpc/21.11`

**Comment by cponder** [comment]
Those are the Fortran *.mod files. Where is the environment module for the esmf, i.e. ```module load ...``` that you were using?

**Comment by cponder** [comment]
And for the 22.1 you used these, right?
```
NVHPC: /glade/u/apps/opt/nvhpc/22.1
ESMF:   /glade/p/cesmdata/cseg/PROGS/esmf/8.3.0b06_casper/openmpi/4.1.1/nvhpc/22.1
```
So it looks like you're using matching components in both cases.
But I'm concerned that the ESMF or some dependency may not have been built correctly.

**Comment by cponder** [comment]
I'm inclined to wait for the NVHPC 22.2 compiler to see if the fixes make any difference.
It should be out later this month.
If there's a problem after that, then I'll dig deeper.

**Comment by sjsprecious** [comment]
> 

Yes, we tried `nvhpc/22.1` as well. The module path is `/glade/p/cesmdata/cseg/PROGS/modulefiles/esmfpkgs/nvhpc/21.11` and `/glade/p/cesmdata/cseg/PROGS/modulefiles/esmfpkgs/nvhpc/22.1`.

**Comment by sjsprecious** [comment]
> I'm inclined to wait for the NVHPC 22.2 compiler to see if the fixes make any difference. It should be out later this month. If there's a problem after that, then I'll dig deeper.

Thanks @cponder .

**Comment by sjsprecious** [comment]
Fixed in #22
